src/test/java/net/imagini/zkcli/CliParametersTest.java

```diff
@@ -1,28 +1,298 @@
 package net.imagini.zkcli;
 
-import net.imagini.zkcli.CliParameters;
-import org.junit.Assert;
+import com.beust.jcommander.ParameterException;
+import net.imagini.jzookeeperedit.ZkClusterManager;
+import org.apache.curator.framework.CuratorFramework;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
 @RunWith(JUnit4.class)
 public class CliParametersTest {
```

Comment:
You're right, what about fewer tests which just test a random combination of options? Similarly for the action/noAction maybe just an example or two rather than testing all of them which is a bit pointless.